easyrest/views/user_controller.py

```diff
@@ -50,3 +55,58 @@ def sign_up(request):
     except IntegrityError:
         database.rollback()
         raise HTTPForbidden("User already exists!")
+
+
+@view_config(route_name='users_list', renderer='json', request_method='GET')
+def get_users_list(request):
+    """This function is intended to display a list of users
+       depending on the role id.
+
+       This function processes the route /users/{role_id}
+       and tries to create a list of users depending on the identifier
+       that is extracted from the parameter {role_id}.
+
+        :param request: GET request
+        :raise HTTPForbidden: If the token is not found
+        :return: Users list, if the user is allowed to view users with the specified role:
+                    {
+                      "message": "Users with role 'role_name'",
+                        "data": [
+                          {
+                            "phone_number": "user_phone_number",
+                            "name": "user_name",
+                            "is_active": is_active_state,
+                            "id": user_id,
+                            "birth_date": "user_birth_date",
+                            "email": "user_email"
+                          }
+                        ],
+                      "success": true,
+                      "error": null
+                    }
+
+                 If the user is not allowed to view the list of users with the specified role:
+                    {
+                      "message": null,
+                      "data": [],
+                      "success": false,
+                      "error": "Action not allowed"
+                    }
+
+    """
+    log = logging.getLogger(__name__)
+    try:
+        current_user = request.token.user
+    except AttributeError as ae:
+        log.error(ae.message)
+        raise HTTPForbidden('Need token for access')
+
+    derivable_role_id = int(request.matchdict['role_id'])
+    role = request.dbsession.query(UserRole).get(derivable_role_id)
+    if role is None:
+        raise HTTPNotFound(request.path)
+    check_action_access(current_user.role.name, foreign_role=role.name, action='read')
+    users_list = [user.as_dict(exclude=['role_id', 'password']) for user in
+                  request.dbsession.query(User).filter_by(role_id=derivable_role_id).order_by(User.name).all()]
+
+    return wrap(users_list, success=True, message='Users with role \'{}\''.format(role.name))
```

Comment:
You may use double quotes to avoid escaping:
```... message="Users with role '{}'".format(...)```

Reply:
Yes, I know about it. But I wanted to stick with common styles. In addition, I wanted to have a response in the format of single quotes like that:
```
{
    "message": "Users with role 'Client'"
}
```